README.md

```diff
@@ -1,207 +1,229 @@
+| accelerator             | [Accelerator Enum](#accelerator-enum) | **REQUIRED.** The intended computational hardware that runs inference.                                                                                                                                                                                                                                                                                                       |
+| accelerator_constrained | boolean                               | **REQUIRED.** True if the intended `accelerator` is the only `accelerator` that can run inference. False if other accelerators, such as amd64 (CPU), can run inference.                                                                                                                                                                                                      |
+| hardware_summary        | string                                | **REQUIRED.** A high level description of the number of accelerators, specific generation of the `accelerator`, or other relevant inference details.                                                                                                                                                                                                                         |
+| container               | [Container](#container)               | **RECOMMENDED.** Information to run the model in a container instance.                                                                                                                                                                                                                                                                                                       |
+| commit_hash             | string                                | Hash value pointing to a specific version of the code used to run model inference. If this is supplied, `source code` should also be supplied and the commit hash must refer to a Git repository linked or described in the `source_code` [Asset Object](stac-asset).                                                                                                        |
+| batch_size_suggestion   | number                                | A suggested batch size for the accelerator and summarized hardware.                                                                                                                                                                                                                                                                                                          |
```

Comment:
After re-reading the doc, I think this should be converted to an actual STAC Asset to better make use of the STAC Item structure. In other words, we would have something like: 

```json
{
   "type": "Feature",
   "properties": {
     "STAC Item Core properites...": {},
     "STAC MLM extension properites...": {},
     "STAC other extension properites...": {}
   },
   "assets": {
     "model": {
       "title": "The model weights.",
       "href": "<URI-to-checkpoint-or-artifact>",
       "type": "application/x-pytorch",
       "roles": ["model", "weights", "data"],
       "mlm:batch_size_suggestion": 64
     },
     "container": {
       "title": "Runtime container for the model.",
       "href": "ghcr.io/my-user/my-repo:latest",
       "type": "application/vnd.oci.image.index.v1+json",
       "roles": ["runtime"],
       "mlm:accelerator": "GPU",
       "mlm:accelerator_constrained": true
     },
     "source": {
       "title": "Source code to work with the model.",
       "href": "https://github.com/my-user/my-repo/tree/main",
       "type": "text/html",
       "roles": ["model", "code", "metadata"]
     }
   }
}
```

According to the STAC Item-Spec, we can have `mlm` extension properties in the specific Assets (https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#additional-fields-for-assets). Those properties can be also under the STAC Item `properties` (which would allow searching/filtering with them). Currently, properties like `mlm:accelerator` cannot be searched because they are nested under `mlm:runtime`. 

Since properties originally nested under `dlm:architecture` (eg: `mlm:total_parameters`, `mlm:framework`, etc.) have been moved upward to the STAC Item `properites` to allow search, we might as well do the same for the `mlm:runtime` when possible.

Reply:
ok I think this makes sense. I'll reorganize and copy relevant mlm:runtime fields upward